src/stores/sdkStore.ts

```diff
@@ -0,0 +1,200 @@
+import { defineStore } from "pinia";
+import {
+  AeSdk,
+  AeSdkAepp,
+  BrowserWindowMessageConnection,
+  CompilerHttp,
+  generateKeyPair,
+  MemoryAccount,
+  Node,
+  SUBSCRIPTION_TYPES,
+  walletDetector,
+} from "@aeternity/aepp-sdk";
+import { COMPILER_URL, nodes } from "../utils/config";
+import { computed, Ref, ref, UnwrapRef } from "vue";
+import BigNumber from "bignumber.js";
+import { getSecretKey, persistSecretKey } from "../utils/utils";
+
+interface Wallet {
+  info: {
+    id: string;
+    type: string;
+    origin: string;
+  };
+  getConnection: () => BrowserWindowMessageConnection;
+}
+interface Wallets {
+  [key: string]: Wallet;
+}
+
+export enum Status {
+  UNINITIALIZED = "Uninitialized",
+  FETCHING_INFO = "Fetching Information",
+  CONNECTED = "Connected",
+  CHECK_FUNDING = "Checking Funding",
+  WALLET_SCANNING = "Wallet Scanning, Please Confirm Connection",
+  CONNECTION_ERROR = "Connection Error",
+}
+
+function getNodes(nodeUrl?: string) {
+  return nodeUrl
+    ? {
+        nodes: [
+          {
+            name: "custom",
+            instance: new Node(nodeUrl!),
+          },
+        ],
+      }
+    : { nodes };
+}
+
+export const useSdkStore = defineStore("sdk", () => {
+  let aeSdk: AeSdkAepp | AeSdk | undefined;
+
+  const secretKey: Ref<UnwrapRef<string | undefined>> = ref();
+
+  const status = ref(Status.UNINITIALIZED);
+  const address = ref();
+  const networkId = ref();
+  const nodeUrl = ref();
+  const isStatic = ref(true);
+
+  const initializedSdk = computed(() => aeSdk);
+
+  async function scanForWallets() {
+    status.value = Status.WALLET_SCANNING;
+
+    return new Promise((resolve, reject) => {
+      if (aeSdk instanceof AeSdkAepp) {
+        const handleWallets: ({
+          wallets,
+          newWallet,
+        }: {
+          wallets: Wallets;
+          newWallet?: Wallet | undefined;
+        }) => Promise<void> = async ({ wallets, newWallet }) => {
+          newWallet = newWallet || Object.values(wallets)[0];
+          stopScan();
+          if (!aeSdk) return;
+
+          try {
+            (aeSdk as AeSdkAepp).disconnectWallet();
+          } catch (_) {}
```

Comment:
reject?

Reply:
no, we specifically ignore, to ensure we are disconnected before trying to connect, but disconnect will error if we are not connected, adding a comment